client/lib/sites-list/list.js

```diff
@@ -222,7 +225,7 @@ SitesList.prototype.update = function( sites ) {
 			//Assign old URL because new url is broken because the site response caches domains
 			//and we have trouble getting over it.
 			if ( site.options.is_automated_transfer && site.URL.match( '.wordpress.com' ) ) {
-				site.URL = siteObj.URL;
+				return siteObj;
```

Comment:
This was designed to assign old URL and prevent URL being changed during small window of transfer before sync is done.
Instead of that, here I return old data so NOTHING is changed in sites list until sync is done